### PR TITLE
BUG: Implement interpolating NaT values in datetime series

### LINKED ISSUE #11701: Interpolate NaT
Hello,

`interpolate` doesn't work with `NaT`
see http://stackoverflow.com/questions/33921795/fill-timestamp-nat-with-a-linear-interpolation/33922824#33922824

Here is a trivial example to show the situation:

```
s = pd.Series(pd.date_range('2015-01-01' , '2015-01-30'), name='t')

s[3], s[4], s[5] = pd.NaT, pd.NaT, pd.NaT
s[13], s[14], s[15] = pd.NaT, pd.NaT, pd.NaT
print(s)

0    2015-01-01
1    2015-01-02
2    2015-01-03
3           NaT
4           NaT
5           NaT
6    2015-01-07
7    2015-01-08
8    2015-01-09
9    2015-01-10
10   2015-01-11
11   2015-01-12
12   2015-01-13
13          NaT
14          NaT
15          NaT
16   2015-01-17
17   2015-01-18
18   2015-01-19
19   2015-01-20
20   2015-01-21
21   2015-01-22
22   2015-01-23
23   2015-01-24
24   2015-01-25
25   2015-01-26
26   2015-01-27
27   2015-01-28
28   2015-01-29
29   2015-01-30
Name: t, dtype: datetime64[ns]

print(s.interpolate())
0    2015-01-01
1    2015-01-02
2    2015-01-03
3           NaT
4           NaT
5           NaT
6    2015-01-07
7    2015-01-08
8    2015-01-09
9    2015-01-10
10   2015-01-11
11   2015-01-12
12   2015-01-13
13          NaT
14          NaT
15          NaT
16   2015-01-17
17   2015-01-18
18   2015-01-19
19   2015-01-20
20   2015-01-21
21   2015-01-22
22   2015-01-23
23   2015-01-24
24   2015-01-25
25   2015-01-26
26   2015-01-27
27   2015-01-28
28   2015-01-29
29   2015-01-30
Name: t, dtype: datetime64[ns]

assert s.interpolate().isnull().sum() == 0
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-150-8a59e397a174> in <module>()
----> 1 assert s.interpolate().isnull().sum() == 0

AssertionError:
```

Kind regards

### PR DESCRIPTION
**Pull request by s-celles**
- [ ] closes #11701
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by scls19fr** [comment]
If I change test from 

        expected = pd.Series(pd.date_range('2015-01-01', '2015-01-30'))

to

        expected = pd.Series(pd.date_range('2015-01-01', '2015-01-30', tz="UTC"))

unit test doesn't pass

this is because of

    if is_datetime64_dtype(self) and self.isnull().any():
        ...

Any idea?

File: pandas/core/series.py [hunk 1]
```diff
@@ -2734,6 +2736,15 @@ def from_csv(cls, path, sep=',', parse_dates=True, header=None,
 
         return result
 
+    def interpolate(self, *args, **kwargs):
+        if (is_datetime64_dtype(self) or
+                is_datetime64tz_dtype(self)) and self.isnull().any():
+            s2 = self.astype('i8').astype('f8')
+            s2[self.isnull()] = np.nan
+            return to_datetime(s2.interpolate(*args, **kwargs))
```
Comment (TomAugspurger) [inline pandas/core/series.py]: I think you'll need to handle timezone information here. Once you do the `.astype('i8')` all TZ info is gone.

If someone has timezones, should it be `tz_convert('UTC') -> interploate -> tz_localize("UTC") -> tz_convert(original)`?

We'll want to test this around DST transitions...

Comment (jreback) [reply pandas/core/series.py]: this should not be done here at all, rather in ``pandas.core.missing.py``

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you rebase an respond to comments.

**Comment by scls19fr** [comment]
I do not have time right now. Sorry
Especially because I think that @TomAugspurger is right!
We need to handle timezone and test around DST transitions carefully
So I'm closing this PR. Please go ahead!

**Pull request closed without merging**